schemas/constructs/component/v1beta1/component.json

```diff
@@ -0,0 +1,115 @@
+      "description": "Format specifies the format used in the `entity.schema` field. JSON is the default."
+    },
+    "model": {
+      "$ref": "https://schemas.meshery.io/v1alpha1/model.json#",
+      "description": "Reference to the specific registered model to which the component belongs and from which model version, category, and other properties may be referenced. Learn more at https://docs.meshery.io/concepts/models"
+    },
+    "metadata": {
+      "type": "object",
+      "description": "Metadata contains additional information associated with the component.",
+      "required": ["name", "version"],
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "Various properties and characteristics of the component."
+        },
+        "capabilities": {
+          "type": "object",
+          "description": "Meshery manages components in accordance with their specific capabilities. This field explicitly identifies those capabilities largely by what actions a given component supports; e.g. metric-scrape, sub-interface, and so on. This field is extensible. ComponentDefinitions may define a broad array of capabilities, which are in-turn dynamically interpretted by Meshery for full lifecycle management."
+        },
+        "genealogy": {
+          "type": "string",
+          "description": "Genealogy represents the various representational states of the component."
+        },
+        "isAnnotation": {
+          "type": "boolean",
+          "description": "Identifies whether the component should be treated as deployable entity or just for logical representation."
+        },
+        "shape": {
+          "type": "string",
+          "description": "Shape of the component used for UI representation.",
+          "enum": ["ellipse","triangle","round-triangle","rectangle","round-rectangle","bottom-round-rectangle","cut-rectangle","barrel","rhomboid","diamond","round-diamond","pentagon","round-pentagon","hexagon","round-hexagon","concave-hexagon","heptagon","round-heptagon","octagon","round-octagon","star","tag","round-tag","vee"]
+        },
+        "primaryColor": {
+          "type": "string",
+          "description": "Primary color of the component used for UI representation."
+        },
+        "secondaryColor": {
+          "type": "string",
+          "description": "Secondary color of the component used for UI representation."
+        },
+        "svgWhite": {
+          "type": "string",
+          "description": "White SVG of the component used for UI representation on dark background."
+        },
+        "svgColor": {
+          "type": "string",
+          "description": "Colored SVG of the component used for UI representation on light background."
+        },
+        "svgComplete": {
+          "type": "string",
+          "description": "Complete SVG of the component used for UI representation, often inclusive of background."
+        },
+        "published": {
```

Comment:
// This should probably becomes status as it is a mutable attribute of the component.

Reply:
and would be an enum of:

ignore
publish
duplicate // this component is a duplicate of another. The component that is to be the canonical reference and that is duplicated by other components should not be assigned the "duplicate" status.
maintenance // unavailable for a period of time